scripts/train_agents.py

```diff
@@ -2,16 +2,200 @@
 mp.set_start_method('spawn', force=True) # should be called before any other module imports
 
 from oai_agents.common.arguments import get_arguments
-from oai_agents.common.tags import TeamType
+from oai_agents.common.tags import TeamType, CheckedPoints
 from oai_agents.common.learner import LearnerType, Learner
 from utils import (get_selfplay_agent_w_tms_collection, 
                 get_fcp_agent_w_tms_clction, 
                 get_eval_types_to_load, 
                 get_fcp_trained_w_selfplay_types, 
                 get_selfplay_agent_trained_w_selfplay_types,
+                get_adversary,
+                get_agent_play_w_adversarys,
                 Curriculum
                 )
 
+def SingleAdversaryPlay(args, 
+                        exp_tag = 'S2FP', 
+                        main_agent_path = None,
+                        main_agent_type = LearnerType.SUPPORTER, 
+                        adversary_type = LearnerType.SELFISHER, 
+                        checked_adversary = CheckedPoints.FINAL_TRAINED_MODEL, 
+                        how_long_init = 4.0,
+                        how_long_for_agent = 1.0,
+                        how_long_for_adv = 1.0,
+                        rounds_of_advplay = 401,
+                        reward_magnifier = 3.0):
+    
+    args.dynamic_reward = False
+    args.final_sparse_r_ratio = 0.5
+    if main_agent_path is None:
+        how_long = how_long_init
+        set_input(args=args, quick_test=quick_test, how_long=how_long)
+        args.SP_seed = args.ADV_seed
+        args.SP_h_dim = args.ADV_h_dim
+        args.learner_type = main_agent_type
+        args.reward_magnifier = reward_magnifier
+        args.exp_dir = exp_tag 
+        SP(args=args, pop_force_training=True)
```

Comment:
regardless of whether `main_agent_path` exists or not, `SP` should be called. Inside of `SP`, you should check for the logic of whether `main_agent_path` exists or not, and `SP` can return the appropriate agent name.

Reply:
Thank you, Ava. This point makes sense in other function. However, in adversary-play, we do not assume that we initialized our agent with self-play agent. We can initialize our agent with a self-play agent, fcp agent, n-1 self-play agent or a previous trained adversary-play agent. Therefore, we don't call SP directly if we already provide a path for them to search which agents to initialize with. 
What do you think?

Reply:
I see. In that case, I think you should have a function that takes care of this process. Something like `def get_initalized_agent(main_agent_path)`, which either returns a SP or the agent loaded from that path.

Reply:
Great advise! I will add that function and rewrite part of the code!